### PR TITLE
allow backslash in string, fixes issue #45

### PR DESCRIPTION
**Pull request by christian-lanius**
This pull request fixes the issue reported in #45 by adding the literal \ (backslash) to the definition of the type STRING. I could not find specifically if "port_instance" in the spec allows for this, but commerical EDA tools support it across the board (and generate such files), thus this library should support it as well.

I have added a testcase in the escaped_name.sdf file, which has crashed before, but has been fixed by the change.

Signed-off-by: Christian Lanius <christian.lanius@rwth-aachen.de>

### CODE REVIEW

**Comment by mithro** [comment]
Hi!

Thank you for your contribution! You will need to rebase this change before we can merge it.

Tim @mithro Ansell

**Review by mithro — COMMENTED** [review]
Looks like some empty `.github/check_license.sh` and `.github/check_python_scripts.sh` files where accidentally added?

**Comment by christian-lanius** [comment]
Yeah, I am currently attempting to figure out what to do with them. They are not empty (as they exist in the repo), but their mode has changed. I'll ping you once I have figured it out.

File: sdf_timing/sdflex.py [hunk 1]
```diff
@@ -150,7 +150,7 @@ def t_SLASH(t):
 
 
 def t_STRING(t):
-    r'[a-zA-Z0-9_\/.\[\]]+'
+    r'[a-zA-Z0-9_\/.\[\]\\\\]+'
```
Comment (kgugala) [inline sdf_timing/sdflex.py:153]: I believe this can be `\\` (not `\\\\`)

Comment (christian-lanius) [reply sdf_timing/sdflex.py:153]: Fixed by commit 3513c57. Not sure why I thought I needed 4, seemed to remember it did not work with 2 only.

**Comment by christian-lanius** [comment]
@kgugala Alright, I fixed the string definition and fixed the file mode for the two scripts which I did not touch.

**Review by kgugala — APPROVED** [review]
LGTM

**Pull request merged**